File: backend/src/main/java/com/team05/petmeeting/domain/shelter/dto/ShelterRes.java
package com.team05.petmeeting.domain.shelter.dto;

import com.team05.petmeeting.domain.shelter.entity.Shelter;

public record ShelterRes (
    String shelterId,  // careRegNo
    String careNm,
    String careTel,
    String careAddr,
    String orgNm
) {
    public static ShelterRes from(Shelter shelter) {
        return new ShelterRes(
            shelter.getCareRegNo(),
            shelter.getCareNm(),
            shelter.getCareTel(),
            shelter.getCareAddr(),
            shelter.getOrgNm()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/shelter/dto/ShelterRes.kt
package com.team05.petmeeting.domain.shelter.dto

import com.team05.petmeeting.domain.shelter.entity.Shelter

data class ShelterRes(
    val shelterId: String,  // careRegNo
    val careNm: String?,
    val careTel: String?,
    val careAddr: String?,
    val orgNm: String?
) {
    companion object {
        @JvmStatic
        fun from(shelter: Shelter): ShelterRes {
            return ShelterRes(
                shelter.careRegNo,
                shelter.careNm,
                shelter.careTel,
                shelter.careAddr,
                shelter.orgNm
            )
        }
    }
}
